cleancat/base.py

```diff
@@ -492,11 +493,20 @@ class Enum(Choices):
     """
     Like Choices, but expects a Python 3 Enum.
     """
-    def __init__(self, enum_cls, choices=None, **kwargs):
-        self.enum_cls = enum_cls
-        if not choices:
-            # Enums are iterable, so we can safely specify choices like that.
-            choices = enum_cls
+    def __init__(self, choices, **kwargs):
+        """Initialize the Enum field.
+
+        The `choices` param can be either:
+        * an enum.Enum class (in which case all of its values will become
+          valid choices),
+        * a list containing a subset of the enum's choices (e.g.
+          `[SomeEnumCls.OptionA, SomeEnumCls.OptionB]`).
+        """
+        is_cls = inspect.isclass(choices)
+        if is_cls:
+            self.enum_cls = choices
+        else:
+            self.enum_cls = choices[0].__class__
```

Comment:
What do we do if it's an empty list? If it's not a required field it should probably just work. Might want to add a note to the docstring that all choices have to belong to the same enum.

Reply:
Done in ddc3cbe.